### PR TITLE
Fetch plugin icons from the WordPress.org API

### PR DESCRIPTION
**Pull request by koke**
This introduces the Remote code to get plugin information from api.wordpress.org and get the icon for each plugin.

Some unusual things in this PR:
- I've used the new Swift 4 `Decodable` and `JSONDecoder` to parse the response. It's pretty straightforward and probably the best way of doing JSON from now on, but it's different from what we do everywhere else.
- Since this is a new API and can't reuse the existing clients. I've used `URLSession` directly. We still have to figure out if we want to start using `Alamofire` or what happens to our network architecture, but since it's just one call for now, I went for the simple barebones solution.

See #7471

To test:

- Load the Plugins section for a site
- Verify that icons are loaded for each plugin that should have one

### CODE REVIEW

File: WordPress/Classes/Stores/PluginStore.swift [hunk 1]
```diff
@@ -217,6 +273,38 @@ private extension PluginStore {
...
+
+    func receivePluginDirectoryEntry(slug: String, entry: PluginDirectoryEntry) {
+        transaction { (state) in
+            state.directoryEntries[slug] = .present(entry)
+            state.fetchingDirectoryEntry[slug] = false
+        }
+    }
+
+    func receivePluginDirectoryEntryFailed(slug: String, error: Error) {
+        // TODO: mark entry as missing if not found
```
Comment (koke) [inline WordPress/Classes/Stores/PluginStore.swift]: I still have to do this so we don't try to fetch info for plugins that we know are not on the directory 😬

**Comment by koke** [comment]
@elibud fixed the TODO item, ready for review.

**Comment by koke** [comment]
Thanks for the review, I'll look into icon loading improvements on another PR

File: WordPressKit/WordPressKit/PluginDirectoryServiceRemote.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import Foundation
+import Alamofire
+
+public struct PluginDirectoryGetInformationEndpoint: Endpoint {
+    static let dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "YYYY-MM-dd h:mma z"
```
Comment (zowesiouff) [inline WordPressKit/WordPressKit/PluginDirectoryServiceRemote.swift]: I came accros this article today: https://app-o-mat.com/post/why-why-why-why mentioning the usage of `YYYY` vs `yyyy`, could it be an issue here? :)

**Pull request merged**